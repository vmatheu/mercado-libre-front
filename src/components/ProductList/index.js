import React from "react";
import { arrayOf, object, string } from "prop-types";
import ProductCard from "../ProductCard";

const ProductList = ({ products, categories }) => {
  return (
    <div>
      <div className="card-header">
        <table>
          <tbody>
            <tr>
              {categories.map((category, index) => (
                <td key={category + index}>
                  {" "}
                  <p className="my-0 font-weight-normal">
                    /{category}{" "}
                  </p>
                </td>
              ))}
            </tr>
          </tbody>
        </table>
      </div>
      <div className="row">
        <div className="mcl-search-product-bar-space" />
      </div>
      <p></p>
      <div className="text-left">
        {products.map((product) => (
          <ProductCard key={product.id} {...product} />
        ))}
      </div>
    </div>
  );
};

ProductList.propTypes = {
  products: arrayOf(object).isRequired,
  categories: arrayOf(string).isRequired,
};

export default ProductList;
