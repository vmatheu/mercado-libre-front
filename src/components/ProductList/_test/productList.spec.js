import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import ProductList from '..';

const buildProduct =
  (id, title, amount, currency, decimals, picture, condition, freeShipping) =>
  ({ id, title, amount, currency, decimals, picture, condition, freeShipping });

describe('<ProductList/>', () => {
  describe('should be list product', () => {
    it('should have the same ui structure with one product', () => {
      const products = [
        buildProduct('test_id_1', 'producto 1', 3000, 'clp', 10, 'uriimage', 'condition', true),
      ];

      const wrapper = shallow(<ProductList products={products} />);
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
    });

    it('should have the same ui structure with five product', () => {
      const products = [
        buildProduct('test_id_1', 'producto 1', 3000, 'clp', 10, 'uriimage', 'condition', true),
        buildProduct('test_id_2', 'producto 2', 3000, 'clp', 10, 'uriimage', 'condition', true),
        buildProduct('test_id_3', 'producto 3', 3000, 'clp', 10, 'uriimage', 'condition', true),
        buildProduct('test_id_4', 'producto 4', 3000, 'clp', 10, 'uriimage', 'condition', true),
        buildProduct('test_id_5', 'producto 5', 3000, 'clp', 10, 'uriimage', 'condition', true),
      ];

      const wrapper = shallow(<ProductList products={products} />);
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
    });
  });
});
