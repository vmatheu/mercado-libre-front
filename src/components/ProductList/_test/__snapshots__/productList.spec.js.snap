// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductList/> should be list product should have the same ui structure with five product 1`] = `
<div>
  <div
    className="card-header"
  >
    <table>
      <tbody>
        <tr>
          <td
            key="categories0"
          >
             
            <p
              className="my-0 font-weight-normal"
            >
              /
              categories
               
            </p>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="row"
  >
    <div
      className="mcl-search-product-bar-space"
    />
  </div>
  <p />
  <div
    className="text-left"
  >
    <ProductCard
      amount={3000}
      condition="condition"
      currency="clp"
      decimals={10}
      freeShipping={true}
      id="test_id_1"
      key="test_id_1"
      picture="uriimage"
      title="producto 1"
    />
    <ProductCard
      amount={3000}
      condition="condition"
      currency="clp"
      decimals={10}
      freeShipping={true}
      id="test_id_2"
      key="test_id_2"
      picture="uriimage"
      title="producto 2"
    />
    <ProductCard
      amount={3000}
      condition="condition"
      currency="clp"
      decimals={10}
      freeShipping={true}
      id="test_id_3"
      key="test_id_3"
      picture="uriimage"
      title="producto 3"
    />
    <ProductCard
      amount={3000}
      condition="condition"
      currency="clp"
      decimals={10}
      freeShipping={true}
      id="test_id_4"
      key="test_id_4"
      picture="uriimage"
      title="producto 4"
    />
    <ProductCard
      amount={3000}
      condition="condition"
      currency="clp"
      decimals={10}
      freeShipping={true}
      id="test_id_5"
      key="test_id_5"
      picture="uriimage"
      title="producto 5"
    />
  </div>
</div>
`;

exports[`<ProductList/> should be list product should have the same ui structure with one product 1`] = `
<div>
  <div
    className="card-header"
  >
    <table>
      <tbody>
        <tr>
          <td
            key="categories0"
          >
             
            <p
              className="my-0 font-weight-normal"
            >
              /
              categories
               
            </p>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="row"
  >
    <div
      className="mcl-search-product-bar-space"
    />
  </div>
  <p />
  <div
    className="text-left"
  >
    <ProductCard
      amount={3000}
      condition="condition"
      currency="clp"
      decimals={10}
      freeShipping={true}
      id="test_id_1"
      key="test_id_1"
      picture="uriimage"
      title="producto 1"
    />
  </div>
</div>
`;
